### PR TITLE
Use Assert.Null when validating against null

### PR DESCRIPTION
**Pull request by jpreese**
Part of #356

### CODE REVIEW

File: generators/Exercise.cs [hunk 1]
```diff
@@ -69,10 +69,19 @@ protected virtual string RenderTestMethodBody(CanonicalDataCase canonicalDataCas
...
+            {
                 return new TestMethodBodyWithExceptionCheck(canonicalDataCase, CanonicalData);
+            }
 
             if (canonicalDataCase.Expected is bool)
+            {
                 return new TestMethodBodyWithBooleanCheck(canonicalDataCase, CanonicalData);
+            }
+
+            if(canonicalDataCase.Expected is null)
```
Comment (robkeim) [inline generators/Exercise.cs]: The Type A in me is screaming about a missing space after the `if` ;)

Comment (robkeim) [inline generators/Exercise.cs]: I forgot that I can fix it inline :)

**Comment by robkeim** [comment]
Nice work @jpreese! The only thing I wonder about this is should we be avoiding `null` at all and just using exceptions. I'll create an issue to discuss it further.

File: generators/Exercise.cs (hunk 1 above)
Comment (jpreese) [inline generators/Exercise.cs]: Oh man, that would bother me too. I didn't even notice that. Usually Visual Studio just automatically tosses that in for me.

**Comment by robkeim** [comment]
Thanks @jpreese 🎉

**Comment by ErikSchierboom** [comment]
> The only thing I wonder about this is should we be avoiding null at all and just using exceptions. I'll create an issue to discuss it further.

👍 I think we should definitely have some discussion on this issue.

**Comment by ErikSchierboom** [comment]
@robkeim Issue created: https://github.com/exercism/csharp/issues/366

**Pull request merged**